testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveCompression.java

```diff
@@ -83,6 +83,34 @@ public void testSnappyCompressedParquetTableCreatedInHive()
         onHive().executeQuery("DROP TABLE " + tableName);
     }
 
+    @Test(groups = HIVE_COMPRESSION)
+    public void testSnappyCompressedLargeCreateTableAsInTrino()
+    {
+        runSnappyCompressedLargeCreateTableAsInTrino(false);
+    }
+
+    @Test(groups = HIVE_COMPRESSION)
+    public void testSnappyCompressedLargeCreateTableAsInTrinoWithNativeWriter()
+    {
+        runSnappyCompressedLargeCreateTableAsInTrino(true);
+    }
+
+    private void runSnappyCompressedLargeCreateTableAsInTrino(boolean optimizedWriter)
+    {
+        String tableName = "large_snappy_compressed_table" + (optimizedWriter ? "_optimized_writer" : "");
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + tableName);
+
+        String catalog = (String) getOnlyElement(getOnlyElement(onTrino().executeQuery("SELECT CURRENT_CATALOG").rows()));
+        onTrino().executeQuery("SET SESSION " + catalog + ".compression_codec = 'SNAPPY'");
+        onTrino().executeQuery("SET SESSION " + catalog + ".experimental_parquet_optimized_writer_enabled = " + optimizedWriter);
+
+        onTrino().executeQuery("CREATE TABLE " + tableName + " WITH (format='PARQUET') AS SELECT orderkey, partkey, suppkey, linenumber FROM tpch.sf5.lineitem");
```

Comment:
wow sf5 is not tiny. how long does the test take, for both writers?

Reply:
It's a couple minutes each. I'll try to replicate the bug with a smaller table tomorrow. The hard part is that the underlying issue had to do with some fallback logic when a Parquet file's dictionary gets too big. The max size for that fallback is configurable, but right now it's hardcoded in Trino.